h2o-core/src/main/java/water/udf/DataChunk.java

```diff
@@ -0,0 +1,26 @@
+package water.udf;
+
+import water.fvec.Chunk;
+import water.fvec.Vec;
+
+/**
+ * Wrapper of a chunk that knows its type, with mutability
+ */
+public abstract class DataChunk<T> implements TypedChunk<T> {
```

Comment:
Yes, it does force boxing. Performance issues should be addressed separately. Having parameterized types does not necessarily imply boxing/unboxing; but so far I did nothing to bypass boxing/unboxing. Now that you have added benchmarking, I guess it may be the right time to check out whether it makes sense; intuitively I hoped to rely on JIT.